lib/ansible/modules/network/checkpoint/cp_network.py

```diff
@@ -0,0 +1,201 @@
+  subnet4:
+    description:
+      - IPv4 network address.
+    type: str
+  subnet6:
+    description:
+      - IPv6 network address.
+    type: str
+  mask_length:
+    description:
+      - IPv4 or IPv6 network mask length. If both masks are required use mask-length4 and mask-length6 fields explicitly
+. Instead of IPv4 mask length it is possible to specify IPv4 mask itself in subnet-mask field.
+    type: int
+  mask_length4:
+    description:
+      - IPv4 network mask length.
+    type: int
+  mask_length6:
+    description:
+      - IPv6 network mask length.
+    type: int
+  subnet_mask:
+    description:
+      - IPv4 network mask.
+    type: str
+  nat_settings:
+    description:
+      - NAT settings.
+    type: dict
+  tags:
+    description:
+      - Collection of tag identifiers.
+    type: list
+  broadcast:
+    description:
+      - Allow broadcast address inclusion.
+    type: str
+    choices: ['disallow', 'allow']
+  color:
+    description:
+      - Color of the object. Should be one of existing colors.
+    type: str
+    choices: ['aquamarine', 'black', 'blue', 'crete blue', 'burlywood', 'cyan', 'dark green', 'khaki', 'orchid', 'dark o
+range', 'dark sea green', 'pink', 'turquoise', 'dark blue', 'firebrick', 'brown', 'forest green', 'gold', 'dark gold', '
+gray', 'dark gray', 'light green', 'lemon chiffon', 'coral', 'sea green', 'sky blue', 'magenta', 'purple', 'slate blue',
+'violet red', 'navy blue', 'olive', 'orange', 'red', 'sienna', 'yellow']
+  comments:
+    description:
+      - Comments string.
+    type: str
+  details_level:
+    description:
+      - The level of detail for some of the fields in the response can vary from showing only the UID value of the objec
+t to a fully detailed representation of the object.
+    type: str
+    choices: ['uid', 'standard', 'full']
+  groups:
+    description:
+      - Collection of group identifiers.
+    type: list
+  ignore_warnings:
+    description:
+      - Apply changes ignoring warnings.
+    type: bool
+  ignore_errors:
+    description:
+      - Apply changes ignoring errors. You won't be able to publish such a changes. If ignore-warnings flag was omitted 
+- warnings will also be ignored.
+    type: bool
+  uid:
+    description:
+      - Object unique identifier.
+    type: str
+  new_name:
+    description:
+      - New name of the object.
+    type: str
+extends_documentation_fragment: checkpoint_objects
+"""
+
+EXAMPLES = """
+- name: add-network
+  cp_network:
+    name: New Network 3
+    nat_settings:
+      auto-rule: true
+      hide-behind: ip-address
+      install-on: All
+      ip-address: 192.0.2.1
+      method: static
+    state: present
+    subnet: 192.0.2.1
+    subnet_mask: 255.255.255.0
+
+- name: set-network
+  cp_network:
+    name : New Network 1
+    new-name : New Network 2
+    color : green
+    subnet : 192.0.0.0
+    mask-length : 16
+    groups : New Group 1
+
+- name: delete-network
+  cp_network:
+    name : New Network 2
```

Comment:
@chkp-orso please include `state` params, in the example. I need to verify with `state: present` the module will call `add-network`, but how can user call `set-network`.
```
state : absent
```

Reply:
@justjais ohh yeah u r right, I will add it.
and as for recognition of `add-network` or `set-network` that's all the logic we implemented in the modules_utils (`checkpoint`)